rac_aspace/serializers.py

```diff
@@ -1,60 +1,38 @@
-import argparse
-from configparser import ConfigParser
-# import csv
-import os
+import csv
 
 
-config = ConfigParser()
-config.read("local_settings.cfg")
+data = {}
 
-file_path = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)), config.get(
-        "Destinations", "filename")
-)
 
-FILE_TYPE_CHOICES = ["csv", "tsv"]
+class Serializer:
 
-column_headings = []
+    def __init__(self, filename, filemode="w"):
+        """
+        Currently this class expects to get the following parameters:
+        Filename: a filename including the file extension
+        Data: Data in a dictionary. Doesn't handle nested arrays.
+        Filemode: Optional argument set to work with already existing files.
+        """
+        self.filename = filename
+        self.fieldnames = list(data.keys())
+        self.data = data
```

Comment:
See comment above

Reply:
Ok, I _think_ I understand what the issue was, but I'm still a little unsure about it. I needed to move the data assignment to the write_data function because that's where we're passing the argument.

I moved fieldnames there too because that's where they're actually being used.

Reply:
Yes - good catch!

Minor point, since you're just setting and using `data` and `fieldnames` in one function, you don't need to set them as properties on the CSVWriter/TSVWriter instance, you can just call them `data` and `fieldnames` rather than `self.data` and `self.fieldnames`.